client/src/java/com/zimbra/client/ZContact.java

```diff
@@ -437,8 +443,31 @@ public InputStream getContentStream() throws ServiceException {
         throw new UnsupportedOperationException("ZContact method not supported yet");
     }
 
+    /**
+     * @return the UID the item is referenced by in the IMAP server.  Returns <tt>0</tt> for items that require
+     * renumbering because of moves.
+     * The "IMAP UID" will be the same as the item ID unless the item has been moved after the mailbox owner's first
+     * IMAP session. */
     @Override
     public int getImapUid() {
-        throw new UnsupportedOperationException("ZContact method not supported yet");
+        if ((null != imapUid) && (imapUid >= 0)) {
+            return imapUid;
+        }
+        if (isGalContact()) {
+            return 0;  /* TODO: is this the best thing to do? */
+        }
+        ZimbraLog.mailbox.debug("ZContact getImapUid() - regetting UID");
+        ZContact zc = null;
+        try {
+            zc = getMailbox().getContact(mId);
```

Comment:
I assumed this worked the same as for messages but I don't think it does.  Needs more research.